src/api/UsersRepository.js

```diff
@@ -62,16 +62,43 @@ export default {
      */
     async addUsernamesToEvents(events) {
         for (let event of events) {
-            if (!sessionStorage[event.userId]) {
+            const storageKey = 'username-' + event.userId;
+            if (!sessionStorage[storageKey]) {
                 try {
                     const userData = await this.getUser(event.userId);
-                    sessionStorage[event.userId] = userData.data.username;
+                    sessionStorage[storageKey] = userData.data.username;
                 } catch (ex) {
                     console.log(ex);
-                    sessionStorage[event.userId] = 'no user found';
+                    sessionStorage[storageKey] = 'no user found';
                 }
```

Comment:
The above changes weren't required, I just wanted to make it more consistent with the new method.